File: archive-romhandbook-ui/lib/types/Formula.ts
export type Formula = {

    id: number

    detail_url: string

    name: string

    formula_code: string
}

export type ApiResponse<T> = {

    success: boolean

    data: T

    meta: unknown
}

export type PaginatedApiResponse<T> = {

    success: boolean

    data: T[]

    meta: {
        page: number
        limit: number
        total: number
        has_next: boolean
    }
}


export type FormulaGraphSummaryItem = {
    node_type: string
    node_count: number
    edge_count: number
}

export type FormulaGraphSummary = {
    formula_id: string
    node_types: FormulaGraphSummaryItem[]
}

export type FormulaGraphNode = {
    node_key: string
    node_type: string
    ref_id: string
    label: string | null
    detail_url: string | null
    image: string | null
    meta_json: string | null
}

export type FormulaGraphEdge = {
    id: number
    from_node_key: string
    to_node_key: string
    edge_type: string
    evidence: string | null
    source_table: string | null
    source_id: string | null
}

export type FormulaGraph = {
    formula_id: string
    depth: number
    nodes: FormulaGraphNode[]
    edges: FormulaGraphEdge[]
}

File: backend-api/internal/models/formula.go
package models

type Formula struct {
	ID string `json:"id"`

	DetailURL string `json:"detail_url"`

	Name string `json:"name"`

	FormulaCode string `json:"formula_code"`
}

type FormulaGraphNode struct {
	NodeKey string `json:"node_key"`

	NodeType string `json:"node_type"`

	RefID string `json:"ref_id"`

	Label *string `json:"label"`

	DetailURL *string `json:"detail_url"`

	Image *string `json:"image"`

	MetaJSON *string `json:"meta_json"`
}

type FormulaGraphEdge struct {
	ID int `json:"id"`

	FromNodeKey string `json:"from_node_key"`

	ToNodeKey string `json:"to_node_key"`

	EdgeType string `json:"edge_type"`

	Evidence *string `json:"evidence"`

	SourceTable *string `json:"source_table"`

	SourceID *string `json:"source_id"`
}

type FormulaGraph struct {
	FormulaID string `json:"formula_id"`

	Depth int `json:"depth"`

	Nodes []FormulaGraphNode `json:"nodes"`

	Edges []FormulaGraphEdge `json:"edges"`
}

type FormulaGraphSummaryItem struct {
	NodeType string `json:"node_type"`

	NodeCount int `json:"node_count"`

	EdgeCount int `json:"edge_count"`
}

type FormulaGraphSummary struct {
	FormulaID string `json:"formula_id"`

	NodeTypes []FormulaGraphSummaryItem `json:"node_types"`
}

type FormulaGraphNodeRelations struct {
	NodeType string `json:"node_type"`

	RefID string `json:"ref_id"`

	Depth int `json:"depth"`

	Center FormulaGraphNode `json:"center"`

	Nodes []FormulaGraphNode `json:"nodes"`

	Edges []FormulaGraphEdge `json:"edges"`
}
